core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -75,6 +75,15 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
   }
 #endif
 #ifdef KOKKOS_ARCH_AMD_GFX942_APU
+  if (!Kokkos::Impl::xnack_enabled()) {
+    std::cerr << R"warning(
+Kokkos::HIP::initialize WARNING: Could not determine that xnack is enabled. 
+                                 Kokkos requires xnack to be enabled for ARCH_AMD_GFX942_APU (MI300A).
```

Comment:
We don't require `xnack` genrally to be enabled anymore since we are only warning.
```suggestion
                                 Kokkos requires xnack to be enabled for ARCH_AMD_GFX942_APU (MI300A) for being able to access host allocations on the device.
```